### PR TITLE
add example for abstractfloat random implementation

### PR DESCRIPTION
**Pull request by udohjeremiah**

### CODE REVIEW

File: stdlib/Random/docs/src/index.md [hunk 1]
```diff
@@ -198,7 +198,28 @@ Given a collection type `S`, it's currently assumed that if `rand(::S)` is defin
 
 #### Generating values for an `AbstractFloat` type
 
-`AbstractFloat` types are special-cased, because by default random values are not produced in the whole type domain, but rather in `[0,1)`. The following method should be implemented for `T <: AbstractFloat`: `Random.rand(::AbstractRNG, ::Random.SamplerTrivial{Random.CloseOpen01{T}})`
+`AbstractFloat` types are special-cased, because by default random values are not produced in the whole type domain, but rather in `[0,1)`. The following method should be implemented for `T <: AbstractFloat`: `Random.rand(::AbstractRNG, ::Random.SamplerTrivial{Random.CloseOpen01{T}})`.
+
+As an example:
+```jldoctest
+julia> struct Point{T} <: AbstractFloat
```
Comment (rfourquet) [inline stdlib/Random/docs/src/index.md:205]: That's somehow surprising to have `Point <: AbstratFloat`.

Comment (udohjeremiah) [reply stdlib/Random/docs/src/index.md:205]: I only used that so the example shows how to maybe generate a random floating-point number for a type with multiple feeds.

Comment (udohjeremiah) [reply stdlib/Random/docs/src/index.md:205]: You have a better suggestion for this?

Comment (udohjeremiah) [reply stdlib/Random/docs/src/index.md]: Check out the current example. I think it is okay.

Comment (fredrikekre) [reply stdlib/Random/docs/src/index.md]: Just make it `MyFloat64` or something, this is IMO equally strange as the `Point` example.

File: stdlib/Random/docs/src/index.md [hunk 2]
```diff
@@ -198,7 +198,28 @@ Given a collection type `S`, it's currently assumed that if `rand(::S)` is defin
...
 #### Generating values for an `AbstractFloat` type
 
-`AbstractFloat` types are special-cased, because by default random values are not produced in the whole type domain, but rather in `[0,1)`. The following method should be implemented for `T <: AbstractFloat`: `Random.rand(::AbstractRNG, ::Random.SamplerTrivial{Random.CloseOpen01{T}})`
+`AbstractFloat` types are special-cased, because by default random values are not produced in the whole type domain, but rather in `[0,1)`. The following method should be implemented for `T <: AbstractFloat`: `Random.rand(::AbstractRNG, ::Random.SamplerTrivial{Random.CloseOpen01{T}})`.
+
+As an example:
+```jldoctest setup = :(Random.seed!(1))
+julia> struct MyFloat <: AbstractFloat
+           x::Float64
+           y::Float64
```
Comment (fredrikekre) [inline stdlib/Random/docs/src/index.md:207]: Just one number, otherwise it is some kind of vector datastructure
```suggestion
```

**Pull request closed without merging**